src/core/geometry/qgsellipse.cpp

```diff
@@ -176,49 +176,85 @@ double QgsEllipse::perimeter() const
 
 QVector<QgsPoint> QgsEllipse::quadrant() const
 {
-  QVector<QgsPoint> quad;
-  quad.append( mCenter.project( mSemiMajorAxis, mAzimuth ) );
-  quad.append( mCenter.project( mSemiMinorAxis, mAzimuth + 90 ) );
-  quad.append( mCenter.project( -mSemiMajorAxis, mAzimuth ) );
-  quad.append( mCenter.project( -mSemiMinorAxis, mAzimuth + 90 ) );
-
-  return quad;
+  return
+  {
+    mCenter.project( mSemiMajorAxis, mAzimuth ),
+    mCenter.project( mSemiMinorAxis, mAzimuth + 90 ),
+    mCenter.project( -mSemiMajorAxis, mAzimuth ),
+    mCenter.project( -mSemiMinorAxis, mAzimuth + 90 )
+  };
 }
 
 QgsPointSequence QgsEllipse::points( unsigned int segments ) const
 {
   QgsPointSequence pts;
 
-  if ( segments < 3 )
+  QVector<double> x;
+  QVector<double> y;
+  QVector<double> z;
+  QVector<double> m;
+
+  pointsInternal( segments, x, y, z, m );
+  const bool hasZ = !z.empty();
+  const bool hasM = !m.empty();
+  pts.reserve( x.size() );
+  for ( int i = 0; i < x.size(); ++i )
   {
-    return pts;
+    pts.append( QgsPoint( x[i], y[i],
+                          hasZ ? z[i] : std::numeric_limits< double >::quiet_NaN(),
+                          hasM ? m[i] : std::numeric_limits< double >::quiet_NaN() ) );
   }
+  return pts;
+}
 
+void QgsEllipse::pointsInternal( unsigned int segments, QVector<double> &x, QVector<double> &y, QVector<double> &z, QVector<double> &m ) const
+{
+  if ( segments < 3 )
+  {
+    return;
+  }
 
-  const Qgis::WkbType pType( mCenter.wkbType() );
-  const double z = mCenter.z();
-  const double m = mCenter.m();
+  const double centerX = mCenter.x();
+  const double centerY = mCenter.y();
+  const double centerZ = mCenter.z();
+  const double centerM = mCenter.m();
+  const bool hasZ = mCenter.is3D();
+  const bool hasM = mCenter.isMeasure();
 
-  QVector<double> t;
-  t.reserve( segments );
-  const double azimuth = std::atan2( quadrant().at( 0 ).y() - mCenter.y(), quadrant().at( 0 ).x() - mCenter.x() );
+  std::vector<double> t( segments );
+  const QgsPoint p1 = mCenter.project( mSemiMajorAxis, mAzimuth );
+  const double azimuth = std::atan2( p1.y() - mCenter.y(), p1.x() - mCenter.x() );
   for ( unsigned int i = 0; i < segments; ++i )
   {
-    t.append( 2 * M_PI - ( ( 2 * M_PI ) / segments * i ) ); // Since the algorithm used rotates in the trigonometric direction (counterclockwise)
+    t[i] = 2 * M_PI - ( ( 2 * M_PI ) / segments * i ); // Since the algorithm used rotates in the trigonometric direction (counterclockwise)
```

Comment:
You're merely carrying the comment, but still, I'm gonna ask :wink: Is that comment missing some context here? I.e., what algorithm?

Reply:
it refers to the ellipse to (segmented) point mathematical algorithm, not a **processing** algorithm